lib/ansible/module_utils/facts/network/iscsi.py

```diff
@@ -79,24 +80,27 @@ def collect(self, module=None, collected_facts=None):
                     iscsi_facts['iscsi_iqn'] = line.split('=', 1)[1]
                     break
         elif sys.platform.startswith('aix'):
-            if module is not None:
-                rc, out, err = module.run_command('/usr/sbin/lsattr -E -l iscsi0 | grep initiator_name')
-                if out:
-                    iscsi_facts['iscsi_iqn'] = out.split()[1].rstrip()
-            else:
-                aixcmd = '/usr/sbin/lsattr -E -l iscsi0 | grep initiator_name'
-                aixret = subprocess.check_output(aixcmd, shell=True)
-                if aixret[0].isalpha():
-                    iscsi_facts['iscsi_iqn'] = aixret.split()[1].rstrip()
+            cmd = get_bin_path('lsattr', required=True)
+            cmd += " -E -l iscsi0"
+            rc, out, err = module.run_command(cmd)
+            if out:
+                line = self.findstr(out, 'initiator_name')
+            iscsi_facts['iscsi_iqn'] = line.split()[1].rstrip()
         elif sys.platform.startswith('hp-ux'):
-            if module is not None:
-                rc, out, err = module.run_command("/opt/iscsi/bin/iscsiutil -l | grep 'Initiator Name'",
-                                                  use_unsafe_shell=True)
-                if out:
-                    iscsi_facts['iscsi_iqn'] = out.split(":", 1)[1].rstrip()
-            else:
-                hpuxcmd = "/opt/iscsi/bin/iscsiutil -l | grep 'Initiator Name'"
-                hpuxret = subprocess.check_output(hpuxcmd, shell=True)
-                if hpuxret[0].isalpha():
-                    iscsi_facts['iscsi_iqn'] = hpuxret.split(":", 1)[1].rstrip()
+            hpuxcmd = "/opt/iscsi/bin/iscsiutil"
+            # try to find it in the default PATH
+            cmd = get_bin_path('iscsiutil')
```

Comment:
Rather than doing the `if not cmd` dance, pass additional search paths to `get_bin_path()`:
```suggestion
            cmd = get_bin_path('iscsiutil', opt_dirs=['/opt/iscsi/bin'], required=True)
```